### ISSUE TITLE
Add back pushlog feature

### ISSUE THREAD

**Issue by whimboo** [mozilla/nightlytt#248]
With the rewrite of the extension we lost this feature. We have to add it back once the WebExtension API allows us to retrieve the changeset id.

**Comment by kyoshino** [comment]
My initial attempt is here, as mentioned in https://github.com/mozilla/nightlytt/issues/235#issuecomment-328309577: 
https://github.com/kyoshino/nightlyttr/compare/experiment

**Comment by whimboo** [comment]
Does it mean that users have to install the experiment addon separately, or how does it work?

**Comment by xabolcs** [comment]
Pushlog is only relevant for short pushlogs.

Experiments run only on [Nightly and Developer Edition.](https://webextensions-experiments.readthedocs.io/en/latest/why.html#where-do-they-work)
Nightly has daily build, Developer Edition has weekly build.

**Comment by kyoshino** [comment]
There are no WebExtensions APIs to retrieve the changeset, crash the app, and achieve other unimplemented features, so I think we need a separate [experiment extension](https://webextensions-experiments.readthedocs.io/en/latest/) anyway. AFAIK the regular extension and experiment cannot be combined.

**Comment by whimboo** [comment]
Maybe it's faster to request the addition of the changeset id to `getBrowserInfo` and `getPlatformInfo`.

**Comment by whimboo** [comment]
The request could be added as dependency for [bug 1215059](https://bugzilla.mozilla.org/show_bug.cgi?id=1215059). Maybe we can even request more data as we need for the customize titlebar feature.

**Comment by xabolcs** [comment]
> Maybe we can even request more data as we need for the customize titlebar feature.

Sure the `prefix` stuff is very limiting.

**Comment by whimboo** [comment]
Which `prefix` stuff do you mean? For the title bar? This is not something I'm referring here for the pushlog feature.

**Comment by xabolcs** [comment]
Yes, I meant the `titlePreface` stuff in `browser.windows.update`. But as You say, it's offtopic here.

**Comment by xabolcs** [comment]
The experiment addon could live on the `experiment` branch, as @kyoshino already did use it.

**Comment by xabolcs** [comment]
> It will be available for the "blueish" Firefox (Nightly / Developer Edition) users (which should be our first class citizen IMHO).

Ohh, there is [Bug 1366827](https://bugzilla.mozilla.org/show_bug.cgi?id=1366827) which could help us for Beta and Release Firefoxes 👍 :

> The main Test Pilot extension and individual Test Pilot experiments are in the process of moving away from the Add-on SDK.  The goal is to use webextension experiments, but of course they'll need those to be usable on release.
> 
> If we land this change in 55, then AMO-signed experiments could theoretically run on release in 55 and 56 but only admins can sign experiments with the AMO CA.  Beginning in 57, only experiments signed with "Mozilla Extensions" will be installable.


@whimboo, would you mind to dig out how this bug could help us?

**Comment by whimboo** [comment]
Sorry but I don't have the time for doing that at the moment. Lets focus on the Nightly and experiment extension first.

**Comment by kyoshino** [comment]
Sorry for the silence. I'm going to work on the 4.1 release in this Q1. I've just heard some good news: [Bug 1323845 - Allow experiments to include content-side APIs](https://bugzilla.mozilla.org/show_bug.cgi?id=1323845). A separated package will no longer be required, so it's a great change for us.

**Comment by whimboo** [comment]
@Loirooriol the Nightly Tester Tools are an open source project and you could have contributed to it instead of creating a new extension. With the code in place we might have found a solution to ship it.
